derive/src/lib.rs

```diff
@@ -69,7 +69,7 @@ pub fn derive_type_def(
                                 path: ident_path(param.ident.clone()),
                             }),
                             colon_token: <Token![:]>::default(),
-                            bounds: std::array::IntoIter::new([
+                            bounds: IntoIterator::into_iter([
```

Comment:
I think `[...].into_iter()` instead of `IntoIterator::into_iter([...])` would be more idiomatic.

Reply:
Tried doing that at first but, it refuse to compile

```rust
  ERROR: a value of type `syn::punctuated::Punctuated<TypeParamBound, syn::token::Add>` cannot be built from an iterator over elements of type `&TypeParamBound`
         the trait `FromIterator<&TypeParamBound>` is not implemented for `syn::punctuated::Punctuated<TypeParamBound, syn::token::Add>`
         the following other types implement trait `FromIterator<A>`:
         <syn::punctuated::Punctuated<T, P> as FromIterator<Pair<T, P>>>
         <syn::punctuated::Punctuated<T, P> as FromIterator<T>>
```

Reply:
Though `vec![..].into_iter().collect()` seems to compile

Reply:
hmm strange, it's fine as-is then